pkg/sources/gcs/gcs.go

```diff
@@ -225,13 +273,46 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) err
 				ctx.Logger().V(1).Info("error setting start progress progress", "name", o.name, "error", err)
 				return
 			}
+			s.setProgress(ctx, o.name, persistableCache)
 		}(o)
 	}
 	wg.Wait()
 
+	s.completeProgress(ctx)
 	return nil
 }
 
+func (s *Source) setupCache(ctx context.Context) *persistableCache {
+	var c cache.Cache
+	if s.Progress.EncodedResumeInfo != "" {
+		c = memory.NewWithData(ctx, strings.Split(s.Progress.EncodedResumeInfo, ","))
+	} else {
+		c = memory.New()
+	}
+
+	// TODO (ahrav): Make this configurable via conn.
+	persistCache := newPersistableCache(defaultCachePersistIncrement, c, &s.Progress)
+	return persistCache
+}
+
+func (s *Source) setProgress(ctx context.Context, objName string, cache cache.Cache) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	ctx.Logger().V(5).Info("setting progress for object", "object-name", objName)
+	s.SectionsCompleted++
+
+	cache.Set(objName, objName)
+	s.Progress.SectionsRemaining = int32(s.stats.numObjects)
+	s.Progress.PercentComplete = int64(float64(s.SectionsCompleted) / float64(s.stats.numObjects) * 100)
```

Comment:
i know this isn't part of this pr so i'm just asking because i'm curious but why does `ProgressComplete` need 64 bits to represent a percentage?

Reply:
It likely doesn't. I think that was the type when the struct was first created a long time ago and we haven't thought twice about it. I can create a separate PR to update that type along with the related code.